runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -5118,172 +5118,260 @@ TR_PrexArgInfo* TR_PrexArgInfo::argInfoFromCaller(TR::Node* callNode, TR_PrexArg
-            }
+         prexArg = new (inliner->trHeapMemory()) TR_PrexArgument(symRef->getKnownObjectIndex(), comp);
+         if (tracePrex)
+            traceMsg(comp, "PREX.inl:      %p: is known object obj%d\n", prexArg, symRef->getKnownObjectIndex());
          }
       else if (argument->getOpCodeValue() == TR::aload)
          {
-         OMR::ParameterSymbol *parmSymbol = argument->getSymbolReference()->getSymbol()->getParmSymbol();
-         if (parmSymbol && !argInfo->get(argOrdinal))
+         OMR::ParameterSymbol *parmSymbol = symbol->getParmSymbol();
+         if (parmSymbol && !prexArg)
             {
             if (parmSymbol->getFixedType())
                {
-               if (priorKnowledge < FIXED_CLASS)
+               prexArg = new (inliner->trHeapMemory()) TR_PrexArgument(TR_PrexArgument::ClassIsFixed, (TR_OpaqueClassBlock *) parmSymbol->getFixedType());
+               if (tracePrex)
                   {
-                  argInfo->set(argOrdinal, new (inliner()->trStackMemory()) TR_PrexArgument(TR_PrexArgument::ClassIsFixed, (TR_OpaqueClassBlock *) parmSymbol->getFixedType()));
-                  if (tracePrex)
-                     {
-                     char *sig = TR::Compiler->cls.classSignature(comp(), (TR_OpaqueClassBlock*)parmSymbol->getFixedType(),trMemory());
-                     traceMsg(comp(), "PREX.inl:      %p: is load of parm with fixed class %p %s\n", argInfo->get(argOrdinal), parmSymbol->getFixedType(), sig);
-                     }
+                  char *sig = TR::Compiler->cls.classSignature(comp, (TR_OpaqueClassBlock*)parmSymbol->getFixedType(), inliner->trMemory());
+                  traceMsg(comp, "PREX.inl:      %p: is load of parm with fixed class %p %s\n", prexArg, parmSymbol->getFixedType(), sig);
                   }
                }
             if (parmSymbol->getIsPreexistent())
                {
-               if (priorKnowledge < PREEXISTENT)
+               int32_t len = 0;
+               const char *sig = parmSymbol->getTypeSignature(len);
+               TR_OpaqueClassBlock *clazz = comp->fe()->getClassFromSignature(sig, len, site->_callerResolvedMethod);
+
+               if (clazz)
                   {
-                  argInfo->set(argOrdinal, new (inliner()->trStackMemory()) TR_PrexArgument(TR_PrexArgument::ClassIsPreexistent));
+                  prexArg = new (inliner->trHeapMemory()) TR_PrexArgument(TR_PrexArgument::ClassIsPreexistent, clazz);
                   if (tracePrex)
-                     traceMsg(comp(), "PREX.inl:      %p: is preexistent\n", argInfo->get(argOrdinal));
+                     traceMsg(comp, "PREX.inl:      %p: is preexistent\n", prexArg);
                   }
                }
             }
+         else if (symbol->isAuto())
+            {
+            TR::Node* valueNode = NULL;
+            defToAutoOrParmInEBB(comp, site->_callNodeTreeTop, symRef, &valueNode);
```

Comment:
Is it worth trying to do this for more than one level of store/load ? i.e. if we had:

```
a1 = parm;

a2 = a1;

load a2;
```

Reply:
Yes, we can. We can keep walking the trees in EBB until we find a termination point, i.e. either we found the object info, or there is no way we can get object info, or we reach the block entry. Will add the change.

Reply:
@vijaysun-omr I modified this function to handle the case you mentioned, could you review again?